contracts/Oxo.sol

```diff
@@ -0,0 +1,172 @@
+pragma solidity 0.4.15;
+contract Oxo {
+
+	enum State { WaitPlayerOne, WaitPlayerTwo, WaitAccept, WinnerPlayerOne, WinnerPlayerTwo, Draw}
+    struct Game{
+		address playerOne;//1
+		address playerTwo;//2
+		uint8[9] board;
+		State state;
+		uint prizeFund;		
+	}
+	
+	uint  public gameId;
+	mapping (uint => Game) public games;
+	event Move(uint _gameId, address playerOne, address playerTwo, State state, string line0, string line1, string line2);
+	event GameOver(State state, string _result);
+	event AcceptGame(string _status);
+	event LogTransfer(address to, uint amount);
+	event BoardOutOrNotEmpty(string Error);
+	event CantMove(string _status, State state);
+	
+	modifier onlyBoardAndEmpty(uint _id, uint _field){
+		if(_field > 8){
+			BoardOutOrNotEmpty("Error: field must be <= 8");
+			return;
+		}
+		if(games[_id].board[_field] != 0){
+			BoardOutOrNotEmpty("Error: field must be empty");
+			return;			
+		}
+		_;
+	}
+
+	modifier onlyCanAccept(uint _prizeFund, uint _id){
+	    if(games[_id].state != State.WaitAccept && msg.sender != games[_id].playerTwo){
+		    return;
+		}
+		if(games[_id].prizeFund/2 > _prizeFund){
+			AcceptGame("Error: msg.value must be = prizeFund/2");
+			return;
+		}
+		_;
+	}
+	
+ 	modifier onlyWaitThisPlayer(uint _id){
```

Comment:
It is not clear from the name of modifier what it's actually doing.
Modifier check do we wait for another player, wait for accepting, check that game hasn't ended but from the name it seems like it just check are we still waiting for player.

Reply:
Firstly, thanks for your time. 🥇 
Really probably you are right, this is not the best name for this modifier. But in my understanding, that if the game is over then she already waits for no one :)